jetty-quickstart/src/main/java/org/eclipse/jetty/quickstart/QuickStartConfiguration.java

```diff
@@ -72,57 +73,70 @@ public static void configureMode(Server server, String mode)
     public enum Mode
     {
         GENERATE,  // Generate quickstart-web.xml and then stop
-        AUTO,      // use or generate depending on the existance of quickstart-web.xml
+        AUTO,      // use or generate depending on the existence of quickstart-web.xml
         QUICKSTART // Use quickstart-web.xml
     }
 
-    private Mode _mode = Mode.AUTO;
-    private boolean _quickStart;
-
     public QuickStartConfiguration()
     {
         super(true);
         addDependencies(WebInfConfiguration.class);
         addDependents(WebXmlConfiguration.class);
     }
 
+    private Mode getModeForContext(WebAppContext context)
+    {
+        Mode mode = (Mode)context.getAttribute(MODE);
+        if (mode == null)
+            return DEFAULT_MODE;
+        return mode;
+    }
+
     @Override
     public void preConfigure(WebAppContext context) throws Exception
     {
-        //check that webapp is suitable for quick start - it is not a packed war
+        // check that webapp is suitable for quick start - it is not a packed war
         String war = context.getWar();
-        if (war == null || war.length() <= 0 || !context.getBaseResource().isDirectory())
-            throw new IllegalStateException("Bad Quickstart location");
+        if (StringUtil.isBlank(war) || !context.getBaseResource().isDirectory())
+            throw new IllegalStateException("Invalid Quickstart location");
 
-        //look for quickstart-web.xml in WEB-INF of webapp
+        // look for quickstart-web.xml in WEB-INF of webapp
         Resource quickStartWebXml = getQuickStartWebXml(context);
-        LOG.debug("quickStartWebXml={} exists={}", quickStartWebXml, quickStartWebXml.exists());
+        if (quickStartWebXml == null)
```

Comment:
This would be a behaviour change. Better to leave the logic as is. Fix up the javadoc for `AUTO` to match the actual documentation here: https://eclipse.dev/jetty/documentation/jetty-11/operations-guide/index.html#og-quickstart